docs/_static/robots.txt

```diff
@@ -0,0 +1,95 @@
+Sitemap: http://docs.soliditylang.org/sitemap.xml
+
+# Prevent development and old documentation from showing up in search results.
+User-agent: *
+Disallow: /en/develop/
+Disallow: /en/breaking/
+
+# Disallow old versions
+Disallow: /en/v0.1.2/
+Disallow: /en/v0.1.3/
+Disallow: /en/v0.1.4/
+Disallow: /en/v0.1.5/
+Disallow: /en/v0.1.6/
+Disallow: /en/v0.1.7/
+Disallow: /en/v0.2.0/
+Disallow: /en/v0.2.1/
+Disallow: /en/v0.2.2/
+Disallow: /en/v0.3.0/
+Disallow: /en/v0.3.1/
+Disallow: /en/v0.3.2/
+Disallow: /en/v0.3.3/
+Disallow: /en/v0.3.4/
+Disallow: /en/v0.3.5/
+Disallow: /en/v0.3.6/
+Disallow: /en/v0.4.0/
+Disallow: /en/v0.4.1/
+Disallow: /en/v0.4.2/
+Disallow: /en/v0.4.3/
+Disallow: /en/v0.4.4/
+Disallow: /en/v0.4.5/
+Disallow: /en/v0.4.6/
+Disallow: /en/v0.4.7/
+Disallow: /en/v0.4.8/
+Disallow: /en/v0.4.9/
+Disallow: /en/v0.4.10/
+Disallow: /en/v0.4.11/
+Disallow: /en/v0.4.12/
+Disallow: /en/v0.4.13/
+Disallow: /en/v0.4.14/
+Disallow: /en/v0.4.15/
+Disallow: /en/v0.4.16/
+Disallow: /en/v0.4.17/
+Disallow: /en/v0.4.18/
+Disallow: /en/v0.4.19/
+Disallow: /en/v0.4.20/
+Disallow: /en/v0.4.21/
+Disallow: /en/v0.4.22/
+Disallow: /en/v0.4.23/
+Disallow: /en/v0.4.24/
+Disallow: /en/v0.4.25/
+Disallow: /en/v0.4.26/
+Disallow: /en/v0.5.0/
+Disallow: /en/v0.5.1/
+Disallow: /en/v0.5.2/
+Disallow: /en/v0.5.3/
+Disallow: /en/v0.5.4/
+Disallow: /en/v0.5.5/
+Disallow: /en/v0.5.6/
+Disallow: /en/v0.5.7/
+Disallow: /en/v0.5.8/
+Disallow: /en/v0.5.9/
+Disallow: /en/v0.5.10/
+Disallow: /en/v0.5.11/
+Disallow: /en/v0.5.12/
+Disallow: /en/v0.5.13/
+Disallow: /en/v0.5.14/
+Disallow: /en/v0.5.15/
+Disallow: /en/v0.5.16/
+Disallow: /en/v0.5.17/
+Disallow: /en/v0.6.0/
+Disallow: /en/v0.6.1/
+Disallow: /en/v0.6.2/
+Disallow: /en/v0.6.3/
+Disallow: /en/v0.6.4/
+Disallow: /en/v0.6.5/
+Disallow: /en/v0.6.6/
+Disallow: /en/v0.6.7/
+Disallow: /en/v0.6.8/
+Disallow: /en/v0.6.9/
+Disallow: /en/v0.6.10/
+Disallow: /en/v0.6.11/
+Disallow: /en/v0.6.12/
+Disallow: /en/v0.7.0/
+Disallow: /en/v0.7.1/
+Disallow: /en/v0.7.2/
+Disallow: /en/v0.7.3/
+Disallow: /en/v0.7.4/
+Disallow: /en/v0.7.5/
+# Allow the last patch release of the 0.7.x series
+#Disallow: /en/v0.7.6/
+Disallow: /en/v0.8.0/
+Disallow: /en/v0.8.1/
+Disallow: /en/v0.8.2/
```

Comment:
Do we really want to disallow these?

Reply:
What would be the point of keeping them indexed?

Reply:
It is a question of what versions do we advocate for. This PR currently says we only advocate the very last releases of the current and the previous series. I'm not sure whether we want that or not, but there should be some agreement about it.

Initially I thought this PR is about getting rid of old links being cached, but old was mentioned as <0.6.0 or <0.7.0 or so.

Reply:
Fair enough. I suggested disabling those versions in https://github.com/ethereum/solidity/pull/10972#discussion_r579345756 and that suggestion has been sitting here for quite a while. When @franzihei asked if there are any blockers to merging this I pointed out this comment and she changed it without a discussion so I thought it wasn't controversial.

So anyway, what is your opinion on this? I think that if the goal is to prevent users from accidentally going to old docs then there's no point in indexing anything but the latest version. The sooner the earlier ones disappear from search results, the better.

Reply:
I thought it was non-controversial since we are talking about the search engine results only here. As an example: If I google for let's say NatSpec Solidity docs I would expect to get shown the current version and not e.g. v0.8.0 just because it got ranked higher in SEO for some reason.

I could still, once I'm in the docs, switch to any other version I like.

Reply:
Wonder why isn't `latest` just a link to the latest version and then we could just allow that as the link?

Reply:
That's an interesting idea. We could just disallow `/en/v0.*/` and let the robot just index `latest`. I don't think it needs to be a redirect for that to work. This way we also won't have to update the file on each release.

Reply:
That's what ``latest``is already currently, or isn't it?

Reply:
Yeah, I guess we'll have to watch the search results and perhaps adjust it afterwards so I think merging this and tweaking it later would be fine.

Reply:
> That's what latestis already currently, or isn't it?

No idea what latest is. It may be equal to develop?

Reply:
It does seem to be equal to v0.8.3. Tangential: I wonder if we could automatically include the version in the generated docs, i.e. "Welcome to the Solidity documentation of 0.8.3"

Reply:
![image](https://user-images.githubusercontent.com/9073706/113327533-60a8b000-931b-11eb-9dc5-151f3fca2cde.png)

Reply:
@chriseth Has that changed recently? I thought that `latest` was the latest release. I even remember @leonardoalt asking recently why he can't see his changes from `develop` on `latest`.

> It does seem to be equal to v0.8.3. Tangential: I wonder if we could automatically include the version in the generated docs, i.e. "Welcome to the Solidity documentation of 0.8.3"

Probably. We do have a commit hash in the footer already.

Also, the config does pull version from `CMakeLists.txt` so Sphinx does have it: https://github.com/ethereum/solidity/blob/develop/docs/conf.py#L62-L74. Might be a matter of just putting it in the page template.

Reply:
To be clear, this discussion is not blocking this PR, but something we probably should do :)